app/src/main/java/com/sasfmlzr/filemanager/MainActivity.java

```diff
@@ -1,69 +1,49 @@
 package com.sasfmlzr.filemanager;
 
-import android.Manifest;
-import android.content.Context;
-import android.content.pm.PackageManager;
+import android.content.Intent;
 import android.os.Bundle;
-import android.os.Environment;
-import android.support.v4.app.ActivityCompat;
-import android.support.v4.content.ContextCompat;
-import android.support.v7.app.AppCompatActivity;
 import android.view.View;
 import android.widget.AdapterView;
 import android.widget.ListView;
-import android.widget.Toast;
-
-import com.sasfmlzr.filemanager.api.adapter.FileExploreAdapter;
-import com.sasfmlzr.filemanager.api.file.FileOperation;
 import com.sasfmlzr.filemanager.api.model.FileModel;
-import com.sasfmlzr.filemanager.api.other.Settings;
-
 import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
-
-public class MainActivity extends AppCompatActivity {
-    protected final int READ_EXTERNAL_STORAGE = 0;
 
+import static com.sasfmlzr.filemanager.api.other.Param.sCountActivity;
+/** View activity*/
+public class MainActivity extends AbstractActivity {
     private ListView mFileList;
-    private FileExploreAdapter mFileExploreAdapter;
-    private final FileOperation mFileOperation = new FileOperation();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        init();
+        mFileList = findViewById(R.id.mFileList);
+        setmFileList(mFileList);
+        Intent intent = getIntent();
+        if(intent.hasExtra("mCurrentPath")){
+            String currentPath = intent.getStringExtra("mCurrentPath");
+            setmFileExploreAdapter(getmFileOperation().loadPath(currentPath, getApplicationContext()));
+            mFileList.setAdapter(getmFileExploreAdapter());
+        }
+        beforeInit(sCountActivity);
+        //Toast.makeText(this, "Main Activity",Toast.LENGTH_SHORT).show();
         AdapterView.OnItemClickListener itemListener = new AdapterView.OnItemClickListener() {
```

Comment:
better either making activity implementing this interface or creating a nested class implementing it